### ISSUE TITLE
Running the `canvas` example causes a log-out

### ISSUE THREAD

**Issue by Phrohdoh** [PistonDevelopers/conrod#1112]
I do not know how better to describe this or what logs to show.

A single-digit number of seconds after the `canvas` example begins running GNOME 3.24.2 crashes and I am sent back to the login view.

Obviously this issue could be anywhere down the stack I just happened to notice this running the mentioned example.

Please let me know what information I can provide to help triage.

**Comment by Phrohdoh** [comment]
I hope this information is helpful.

When reviewing the syslog after the mentioned crash I see this line:

```
Dec 21 08:42:49 txps org.gnome.Shell.desktop[1952]: mutter:ERROR:wayland/meta-wayland-surface.c:1192:update_surface_output_state: assertion failed: (surface->role)
```

**Comment by mitchmindtree** [comment]
Thanks for the issue!

Does this happen with any of the other conrod examples? I'm guessing it might happen with all examples that use the `winit` feature, however if it does not that would be very useful.

I think that this will likely have more to do with the wayland [winit](https://github.com/tomaka/winit) backend, or possibly even the mutter wayland implementation (I've had to patch a bug there once before). Someone in gnome's #mutter IRC might be able to point us in the right direction, they were very helpful when I needed to get a mutter dev environment running.

I've (sadly) switched back to using X for the past few months, simply because it's a bit more stable and many of the applications I use don't properly handle HiDPI (I'm not sure if it's possible to through xwayland?) which is a big problem on my 12" 4k laptop screen. That said, it's been a few months since I tried it and I'm not sure how far it has progressed!

@vberger have you run into this error before?

**Comment by vberger** [comment]
I'm not a gnome user myself, so I didn't see this kind of error myself. It looks like for some reason mutter tries to display a surface without a role... ?

Anyway, on this that I can tell for sure, is that it is a mutter bug. Even if the conrod app is doing something wrong from the wayland protocol point of view, mutter should not crash like that, but rather raise a protocol error and kill the client (the conrod app). So imo the best course of action for now would be to open a bug report to mutter.

**Comment by cellscape** [comment]
It is indeed bug in mutter and it seems it was fixed in mutter 3.28: https://gitlab.gnome.org/GNOME/mutter/commit/07745b3082d4442fba2377c11f2082c85dd8f2af. At least manually applying part of patch on mutter 3.24.4 fixes logout caused by gnome-shell crash for me.
